packages/language-core/src/generators/template.ts

```diff
@@ -1224,6 +1224,7 @@ export function generate(
 		codes.push(`}, `);
 
 		for (const prop of props) {
+			const shouldCamelize = !nativeTags.has(node.tag) || node.tag === 'component' || node.tag === 'Component';
```

Comment:
Feels hacky since the element can be called anything really. It can be a `div` even.

Wouldn't it make sense to force camelize when `node.tagType === CompilerDOM.ElementTypes.COMPONENT`? It appears to be a component type in this case.

Reply:
`nativeTags` is used in the case of using vue-language-tools for projects other than vue, e.g. uni-app projects need to specify `nativeTags` to ignore the original tag names: https://github.com/uni-helper/uni-app-types#%E9%85%8D%E7%BD%AE

Originally #3750 judgment based on `tagType` would break this use case.

I plan to remove `nativeTags` from the separate PR and suggest users to modify template compiler options via plugin api to replace `nativeTags` option, so it should be possible to change to fully `tagType` based judgment.